sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -591,18 +591,40 @@ case class MapObjects private(
       case _ => inputData.dataType
     }
 
-    val (getLength, getLoopVar) = inputDataType match {
+    val (getLength, prepareLoop, getLoopVar) = inputDataType match {
       case ObjectType(cls) if classOf[Seq[_]].isAssignableFrom(cls) =>
```

Comment:
Also not sure if you want to change this here, but think this is more direct as `cls.isInstanceOf[Seq[_]]`

Reply:
you can use `seqObj.isInstanceOf[Seq[_]]`, but not a `Class`

Reply:
Oops, right, not reading this clearly.